src/com/chess/Game/ChessBoard.java

```diff
@@ -53,81 +53,95 @@ protected ChessBoard(Piece[][] board, NamedColor p1Color, NamedColor p2Color) {
+        if (!forCheck) {
+            List<Move> validMoves = new ArrayList<>();
+            for (Move m: moves) {
+                ChessBoard clone = clone();
+                clone.makeMove(m);
+                if ((m.isCapture() && getPiece(m.getCapture()) instanceof King) || !clone.isChecked(p.getColor())) {
+                    validMoves.add(m);
+                }
+            }
+            return validMoves;
+        }
         return moves;
     }
 
-    public boolean makeMove(Move move) {
-        return makeMove(move, false);
-    }
-
-    public boolean makeMove(Move move, boolean force) {
-        Coordinate from = move.getFrom();
-        if (!force) {
-            List<Move> validMoves = getAllMoves(from);
-
-            boolean valid = false;
-            for (Move m : validMoves) {
-                if (move.equals(m)) {
-                    valid = true;
+    public boolean isChecked(NamedColor player) {
+        // Find king for that player
+        Coordinate kingPosition = null;
+        for (int x = 0; x < board.length; x++) {
+            for (int y = 0; y < board[x].length; y++) {
+                Coordinate test = new Coordinate(x, y);
+                Piece p = getPiece(test);
+                if (p != null && p instanceof King && p.getColor().equals(player)) {
+                    kingPosition = test;
+                    break;
                 }
             }
-            if (!valid) {
-                return false;
+        }
+        // check if the kings position is under threat
+        if (kingPosition == null) {
+            return false;
+        }
+        return isUnderThreat(kingPosition, player);
+    }
+
+    public boolean isUnderThreat(Coordinate position, NamedColor player) {
+        ChessBoard clone = clone();
+        NamedColor enemy;
+        int p;
+        if (p1Color.equals(player)) {
+            enemy = p2Color;
+            p = 2;
+        } else {
+            enemy = p1Color;
+            p = 1;
+        }
+        clone.setPosition(position, new Pawn(player, p));
+        List<Move> enemyMoves = clone.getAllMoves(enemy, true);
+        for (Move m: enemyMoves) {
+            if (m.isCapture() && m.getCapture().equals(position)) {
+                return true;
             }
         }
+        return false;
+    }
 
+    public boolean makeMove(Move move) {
+        Coordinate from = move.getFrom();
         Piece p = setPosition(from, null);
         if (move.isCapture()) {
             setPosition(move.capture, null);
         }
         setPosition(move.getTo(), p);
         p.onMove(move);
         if (move.getOtherMove() != null) {
-            return makeMove(move.getOtherMove(), true);
+            return makeMove(move.getOtherMove());
         }
         return true;
     }
 
-    /**
-     * TODO need to update to do checkmates
-     * @return null if the game is not over. otherwise the color of the winner
-     */
-    public NamedColor isGameOver() {

```

Comment:
Don't really need this since it is the same as no moves and check but could be persuaded to add it back